xray/plot/plot.py

```diff
@@ -257,9 +260,34 @@ def _determine_cmap_params(plot_data, vmin=None, vmax=None, cmap=None,
     return vmin, vmax, cmap, extend
 
 
+# MUST run before any 2d plotting functions are defined since
+# _plot2d decorator adds them as methods here.
```

Comment:
This is the best solution I've come up with for being able to use `DataArray.plot.<plotfunc>`. If there's a better way to do it I'm all ears.